sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/signaling/SignalingClient.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.chat.signaling;
+
+/**
+ * The signaling client interface
+ */
+public interface SignalingClient {
+
+    /**
+     * flag to indicate if signaling client has started
+     * @return boolean if signaling client has started
+     */
+    boolean hasStarted();
+
+    /**
+     * Start the realtime connection.
+     */
+    void start();
+    /**
+     * Stop the realtime connection and unsubscribe all event handlers.
+     */
+    void stop();
+
+    /**
+     * Listen to Chat events.
+     * @param chatEventId the chat event id
+     * @param listenerId a listener id that is used to identify the listner
+     * @param listener the listener callback function
+     */
+    void on(String chatEventId, String listenerId, RealTimeNotificationCallback listener);
```

Comment:
Under `properties` package there is a type named `ChatEventId`, should we take that type as first parameter instead of plain string? plain string forces users to lookup documentation to find the right value. If the reason for accepting string is to future proof new values, then we can explore making `ChatEventId` an extensible enum? (like [this](https://github.com/Azure/azure-sdk-for-android/blob/945e07b46c5db0a5220512f5e6c31cb815db92c3/sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationCloudEnvironment.java#L10)). Please see if that's a possible option. Also wondering we can get rid of `properties` sub package and have that type in the `signaling` package or under `chat.models` package.

Reply:
This is to be consistent with other languages, we take a string chatEventId in on and off functions